pci/src/configuration.rs

```diff
@@ -21,7 +21,7 @@ const ROM_BAR_REG: usize = 12;
 const BAR_IO_ADDR_MASK: u32 = 0xffff_fffc;
 const BAR_MEM_ADDR_MASK: u32 = 0xffff_fff0;
 const ROM_BAR_ADDR_MASK: u32 = 0xffff_f800;
-const MSIX_REGISTER_MASK: u32 = 0x471_0000;
+const MSIX_REGISTER_MASK: u32 = 0x3fff_0000;
```

Comment:
A few things:
- If I read the code correctly, the `writable_bits` is about the bits that we will let the guest write. Therefore the mask should be `0xc0000000`.
- I would change the name of this variable so that it makes it very clear that it relates to the MSI-X message control.
- Shouldn't we do the same for MSI?

Reply:
> A few things:
> 
>     * If I read the code correctly, the `writable_bits` is about the bits that we will let the guest write. Therefore the mask should be `0xc0000000`.

Yup, I got this wrong
> 
>     * I would change the name of this variable so that it makes it very clear that it relates to the MSI-X message control.
> 

Sounds good.

>     * Shouldn't we do the same for MSI?

Do we support MSI?

Reply:
> > * Shouldn't we do the same for MSI?
> > ```
> 
> Do we support MSI?

The configuration.rs code never seems to do anything about MSI.

Reply:
I know because nothing's really needed from MSI, but it'd be good to apply the patch you wrote for MSI.

Reply:
@sboeuf The masking infrastructure is only in configuration.rs and only needs to be there because of the saved values. If MSI doesn't have a register in that space then it doesn't matter. The `MsixConfig` and `MsiConfig` structures have the correct handling of the bits.

Reply:
Well MSI has a capability structure as well. So the point is that if the guest tries to write to it, it should discard the bits that are not writable.

Reply:
> Well MSI has a capability structure with 16bits message control as well. So the point is that if the guest tries to write to it, we should discard the bits that are not writable.

If might be easier if you just push a patch onto the PR as I don't think I can see it.

Reply:
Go ahead and merge this PR, I'll send a follow up one.